pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/RulesToExecute.java

```diff
@@ -134,15 +143,20 @@ public boolean isRecommendBloomFilter() {
     return _recommendBloomFilter;
   }
 
+  public boolean isRecommendRealtimeProvisioning() {
+    return _recommendRealtimeProvisioning;
+  }
+
   // Be careful with the sequence, each rule can execute individually
   // but a rule may depend on its previous rule when they both fired
   public enum Rule {
-    FlagQueryRule, 
+    FlagQueryRule,
     KafkaPartitionRule,
     InvertedSortedIndexJointRule,
     NoDictionaryOnHeapDictionaryJointRule, // NoDictionaryOnHeapDictionaryJointRule must go after InvertedSortedIndexJointRule since we do not recommend NoDictionary on cols with indices
     VariedLengthDictionaryRule, // VariedLengthDictionaryRule must go after NoDictionaryOnHeapDictionaryJointRule  since we do not recommend dictionary on NoDictionary cols
     PinotTablePartitionRule, // PinotTablePartitionRule must go after KafkaPartitionRule to recommend realtime partitions, after NoDictionaryOnHeapDictionaryJointRule to correctly calculate record size
     BloomFilterRule,
+    RealtimeProvisioningRule
```

Comment:
Add comments here please explaining the new rule.
What would be better is to move these comments as javadoc for the enum and describe each rule in a bullet list. Will look much cleaner IMO.

Reply:
Each Rule class has its own javadoc explaining what they do. The comments here are for the order of the rules. Good that you pointed this out, I need to add a comment here explaining that this rule needs to be executed as the last rule as it uses the output of the other rules as its input.